File: server/app/schemas/photo.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime, date
from enum import Enum
class PhotoStatus(str, Enum):
    public = "public"
    private = "private"
    archived = "archived"
    draft = "draft"
class PhotoBase(BaseModel):
    title: Optional[str] = None
    slug: Optional[str] = None
    description: Optional[str] = None
    image_url: str
    taken_at: Optional[date] = None
    location: Optional[str] = None
    status: Optional[PhotoStatus] = PhotoStatus.draft

class PhotoCreate(PhotoBase):
    album_id: Optional[int] = None
    user_id: Optional[int] = None
    tag_ids: Optional[List[int]] = []

class PhotoResponse(PhotoBase):
    id: int
    album_id: Optional[int]
    user_id: Optional[int]
    created_at: datetime
    tags: List[str] = []

    class Config:
        from_attributes = True

File: client/types/photo.ts
// client/types/photo.ts

export interface Photo {
  id: number;
  title: string;
  slug: string;
  description?: string;
  image_url?: string;
  taken_at?: Date | null;
  location?: string;
  album_id?: number | null;
  user_id?: number | null;
  order?: number;
  created_at?: Date | null;
  status: string;
}

// Kiểu dữ liệu cho Form Data (có thể chứa File khi upload)
export interface PhotoFormData extends Omit<Partial<Photo>, 'image_url'> {
    // cover_image có thể là File khi upload, hoặc string URL khi ở chế độ edit
    image_url?: string | File | File[];
}

// Các trạng thái cơ bản (không bao gồm các hàm)
export interface PhotoBaseState {
  photos: Photo[];
  formData: PhotoFormData;
  editingPhoto: Photo | null;
  isModalOpen: boolean;
  isUploading: boolean;
  uploadedPhotoId: number | null;

  // Phân trang backend
  currentPage: number;
  totalPages: number;
  itemsPerPage: number;
  totalItems: number;
}